tests/bluetooth/bsim_bt/bsim_test_eatt/src/main_collision.c

```diff
@@ -0,0 +1,221 @@
+	}
+
+	/* Wait for EATT channels to be connected */
+	k_sleep(K_MSEC(10000));
+	num_enhanced = bt_att_num_enhanced_bearers(default_conn);
+	printk("%d enhanced bearers connected\n", num_enhanced);
+	if (num_enhanced != CONFIG_BT_EATT_MAX) {
+		FAIL("Expected %d enhanced bearers, got %d\n", CONFIG_BT_EATT_MAX, num_enhanced);
+	}
+	/* Disconnect */
+	err = bt_conn_disconnect(default_conn, BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+	if (err) {
+		FAIL("Disconnection failed (err %d)\n", err);
+	}
+
+	while (is_connected) {
+		k_sleep(K_MSEC(100));
+	}
+
+	PASS("EATT Peripheral tests Passed\n");
+}
+
+static void device_found(const bt_addr_le_t *addr, int8_t rssi, uint8_t type,
+			 struct net_buf_simple *ad)
+{
+	struct bt_le_conn_param *param;
+	int err;
+
+	err = bt_le_scan_stop();
+	if (err) {
+		FAIL("Stop LE scan failed (err %d)\n", err);
+	}
+
+	param = BT_LE_CONN_PARAM_DEFAULT;
+	err = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, param, &default_conn);
+
+	if (err) {
+		FAIL("Create conn failed (err %d)\n", err);
+	}
+	printk("Device connected\n");
+}
+
+static void test_central_main(void)
+{
+	struct bt_le_scan_param scan_param = {
+		.type = BT_LE_SCAN_TYPE_ACTIVE,
+		.options = BT_LE_SCAN_OPT_NONE,
+		.interval = BT_GAP_SCAN_FAST_INTERVAL,
+		.window = BT_GAP_SCAN_FAST_WINDOW,
+	};
+
+	int err;
+	uint16_t num_enhanced;
+
+	err = bt_enable(NULL);
+	if (err) {
+		FAIL("Can't enable Bluetooth (err %d)\n", err);
+	}
+
+	err = bt_le_scan_start(&scan_param, device_found);
+	if (err) {
+		FAIL("Scanning failed to start (err %d)\n", err);
+	}
+
+	while (!is_connected) {
+		k_sleep(K_MSEC(100));
+	}
+
+	err = bt_eatt_connect(default_conn, CONFIG_BT_EATT_MAX);
+	if (err) {
+		FAIL("Sending credit based connection request failed (err %d)\n", err);
+	}
+
+	/* Wait for EATT channels to be connected */
+	k_sleep(K_MSEC(10000));
+	num_enhanced = bt_att_num_enhanced_bearers(default_conn);
+	printk("%d enhanced bearers connected\n", num_enhanced);
+	if (num_enhanced != CONFIG_BT_EATT_MAX) {
+		FAIL("Expected %d enhanced bearers, got %d\n", CONFIG_BT_EATT_MAX, num_enhanced);
+	}
+
+	/* Wait for disconnect */
+	while (is_connected) {
+		k_sleep(K_MSEC(100));
+	}
+
+	PASS("EATT Central tests Passed\n");
+}
+
+static void test_init(void)
+{
+	bst_ticker_set_next_tick_absolute(60e6); /* 60 seconds */
+	bst_result = In_progress;
+}
+
+static void test_tick(bs_time_t HW_device_time)
+{
+}
+
+static const struct bst_test_instance test_def[] = { { .test_id = "peripheral",
+						       .test_descr = "Peripheral Collision",
+						       .test_post_init_f = test_init,
+						       .test_tick_f = test_tick,
+						       .test_main_f = test_peripheral_main },
+						     { .test_id = "central",
+						       .test_descr = "Central Collision",
+						       .test_post_init_f = test_init,
+						       .test_tick_f = test_tick,
+						       .test_main_f = test_central_main },
+						     BSTEST_END_MARKER };
```

Comment:
I think this is more readable: 

```c
static const struct bst_test_instance test_def[] = { 
	{ 
		.test_id = "peripheral",
		.test_descr = "Peripheral Collision",
		.test_post_init_f = test_init,
		.test_tick_f = test_tick,
		.test_main_f = test_peripheral_main 
	},
	{ 
		.test_id = "central",
		.test_descr = "Central Collision",
		.test_post_init_f = test_init,
		.test_tick_f = test_tick,
		.test_main_f = test_central_main 
	},
	BSTEST_END_MARKER 
};
```

Reply:
Auto-format wanted it the way I had it. Agree your suggestion is better.

Reply:
Looks like it wasn't updated